nautobot/extras/tests/integration/test_plugins.py

```diff
@@ -0,0 +1,107 @@
+import os
+import re
+from unittest import skipIf
+
+from django.conf import settings
+from django.contrib.contenttypes.models import ContentType
+from django.urls import reverse
+
+
+from nautobot.extras.choices import WebhookHttpMethodChoices
+from nautobot.extras.context_managers import web_request_context
+from nautobot.extras.models import Webhook
+from nautobot.users.models import Token
+from nautobot.utilities.testing.integration import SplinterTestCase
+
+from dummy_plugin.models import DummyModel
+
+
+@skipIf(
+    "dummy_plugin" not in settings.PLUGINS,
+    "dummy_plugin not in settings.PLUGINS",
+)
+class PluginWebhookTest(SplinterTestCase):
+    """
+    This test case proves that plugins can use the webhook functions when making changes on a model.
+
+    Because webhooks use celery a class variable is set to True called `requires_celery`. This starts
+    a celery instance in a separate thread.
+    """
+
+    requires_celery = True
+
+    def setUp(self):
+        super().setUp()
+        for f in os.listdir("/tmp"):
+            if re.search("test_plugin_webhook_.+", f):
+                os.remove(os.path("/tmp", f))
+
+        self.url = f"http://localhost:{self.server_thread.port}" + reverse(
+            "plugins-api:dummy_plugin-api:dummymodel_webhook"
+        )
+        self.login(self.user.username, self.password)
```

Comment:
Doesn't seem like logging in through the UI should be necessary for the purposes of this test.